tasks/managed/update-trusted-tasks/update-trusted-tasks.yaml

```diff
@@ -0,0 +1,91 @@
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to update trusted-tasks list bundles.
+    Once a task is released as a bundle, the list of trusted-tasks should be 
+    updated with the new task.
+    If an OCI artifact named data-acceptable-bundles is already in place we will use
+    it as an input and append it, else we will create a new artifact
+  params:
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot json reside
+  steps:
+    - name: update-trusted-tasks
+      image: quay.io/konflux-ci/appstudio-utils@sha256:0901e46d9a41d0c564b0efab092f37cd187bd8b4ff2455ec84539ceb8b08f3ea
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+        
+        SNAPSHOT_SPEC_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        TAG="latest"
+
+        # check if snapshot exsits
+        if [ ! -f "${SNAPSHOT_SPEC_FILE}" ] ; then
+            echo "No valid snapshot file was found."
+            exit 1
+        fi
+
+        # Extract the application from the snapshot
+        application=$(jq -r '.application' "${SNAPSHOT_SPEC_FILE}")
+
+        # Get the number of components
+        NUM_COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
+
+        printf 'Beginning "%s" for "%s"\n\n' "$(context.task.name)" "$application"
+        for ((i = 0; i < NUM_COMPONENTS; i++))
+        do
+            # extract the component from the snapshot 
+            component=$(jq -c --argjson i "$i" '.components[$i]' "${SNAPSHOT_SPEC_FILE}")
+
+            # Extract the repository from the snapshot
+            repository=$(jq -r '.repository' <<< "$component")
+
+            # The data-acceptable-bundles repo should reside in the same org in the registry.
+            # Fetch the repository and replace the repo with data-acceptable-bundles
+            # for example: quay.io/myorg/myrepo -> quay.io/myorg/data-acceptable-bundles
+            ACCEPTABLE_BUNDLES=$(echo "${repository}" | awk -F'/' '{$NF="data-acceptable-bundles"; print $0}' OFS='/')
+
+            # Extract the image sha
+            sha=$(jq -r '.containerImage' <<< "$component" | awk -F'@' '{print "@" $2}')
+            
+            # Get the number of tags from the component
+            NUM_TAGS=$(jq '.tags | length' <<< "$component")
+
+            # update the tasks list OCI artifact for each tag in the component
+            for ((j = 0; j < NUM_TAGS; j++))
+            do
+                imageTag=$(jq -c -r --argjson j "$j" '.tags[$j]' <<< "$component")
+                
+                set +e
+                # Check if ACCEPTABLE_BUNDLES OCI artifact has a latest tag
+                skopeo list-tags docker://"${ACCEPTABLE_BUNDLES}" | jq -r '.Tags[]' | grep ${TAG} &> /dev/null
```

Comment:
`ACCEPTABLE_BUNDLES` is set once per component. It seems like a waste to do this same request repeatedly for each tag. It should be done once before this for loop and the output saved to a tmp file and here you would just do the grep. But also, you should be careful about partial matches - the grep should use something like `"^${TAG}$"`.

Reply:
I was considering the following scenario:
Component with multiple tags with the same `ACCEPTABLE_BUNDLES` url and `ACCEPTABLE_BUNDLES`  does not exist.
For the first tag skopeo will return error (the tag does not exist), and will run the `ec track bundle` command without the input parameter. 
However, for the second tag in the same component, skopeo will return 0 because we already created the `ACCEPTABLE_BUNDLES` in the previous tag, and therefor will use the `ec track bundle` command with the input parameter. 
In addition, I was thinking that most of the components will only have one tag.

Reply:
The `ACCEPTABLE_BUNDLES` is set only once per component and not for each tag:
At line 59:
ACCEPTABLE_BUNDLES=$(echo "${repository}" | awk -F'/' '{$NF="data-acceptable-bundles"; print $0}' OFS='/')

The skopeo call to check if it exists or not is done per tag

Reply:
Oh, so `ec track bundle` will push the $ACCEPTABLE_BUNDLES:$TAG image so that on the next tag it will already exist? I did not realize that. And given that most of the time there will be just one tag, as you say, it should be ok.

Reply:
Exactly @mmalina .
I also updated `skopeo list-tags docker://"${ACCEPTABLE_BUNDLES}" | jq -r '.Tags[]' | grep ${TAG} &> /dev/null`
to `skopeo list-tags docker://"${ACCEPTABLE_BUNDLES}" | jq -r '.Tags[]' | grep "^${TAG}$" &> /dev/null`
As you suggested

Reply:
Thanks. I assumed the tag list would be a tag per line (I should have known better, given that you already had the `jq` there). So this won't work - see the new thread I created.